crates/relayer/relayer-components/src/chain/traits/types/client_state.rs

```diff
@@ -6,7 +8,7 @@ pub trait HasClientStateType<Counterparty>: Async {
     /**
         The client state of the `Self` chain's client on the `Counterparty` chain
     */
-    type ClientState: Async;
+    type ClientState: Debug + Async;
```

Comment:
In general, it is better to introduce trait bounds on implementation side, instead of on the trait.

You can do that by adding `Chain::ClientState: Debug` to the implementation that needs the constraint.

We would only add the trait bound to the trait associated type if it is used in almost every implementation.